java/code/src/com/redhat/rhn/frontend/xmlrpc/LoggingInvocationProcessor.java

```diff
@@ -82,33 +121,68 @@ public boolean before(XmlRpcInvocation invocation) {
      * {@inheritDoc}
      */
     public Object after(XmlRpcInvocation invocation, Object returnValue) {
-        StringBuffer buf = new StringBuffer();
+        StringBuffer arguments = processArguments(
+            invocation.getHandlerName(),
+            invocation.getMethodName(),
+            invocation.getArguments()
+        );
+        afterProcess(
+            invocation.getHandlerName(),
+            invocation.getMethodName(),
+            arguments,
+            RhnXmlRpcServer.getCallerIp(),
+            getCallerLogin(),
+            getStopWatch()
+        );
+
+        return returnValue;
+    }
+
+    /**
+     * Makes the after hook processing logic agnostic, taking as parameters all data to be logged and
+     * the processingTimer to be stopped.
+     *
+     * @param handlerName the handler of the request
+     * @param methodName the method called
+     * @param arguments the string representation of the arguments passed to the method
+     * @param ip the request's ip
+     * @param callerLogin the user's login
+     * @param processingTimer to stop and get the processing time.
+     */
+    public static void afterProcess(
+        String handlerName,
+        String methodName,
+        StringBuffer arguments,
+        String ip,
+        String callerLogin,
+        StopWatch processingTimer
```

Comment:
while this way is convenient, stopping the timer logically should not be a part of this method. even though this method is moved to the base class, timer operation should be left to the subclasses.